Changelog command uses deprecated `git whatchanged` leading to flaky Tests on Mac OS X

### Affected version

HEAD

### Bug description

Tests look broken or at least flaky:

```
Error:  org.apache.maven.scm.provider.git.gitexe.command.changelog.GitExeChangeLogCommandTckTest.testChangeLogCommand -- Time elapsed: 1.260 s <<< FAILURE!
java.lang.AssertionError: 
The git-log command failed.: /bin/sh -c cd '/Users/runner/work/maven-scm/maven-scm/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/target/scm-test/working-copy' && 'git' 'whatchanged' '--format=medium' '--decorate=short' '--raw' '--no-merges' '--date=iso' '--' '.'
'git whatchanged' is nominated for removal.
If you still use this command, please add an extra
option, '--i-still-use-this', on the command line
and let us know you still use it by sending an e-mail
to <git@vger.kernel.org>.  Thanks.
fatal: refusing to run without --i-still-use-this
```

seems to be reproducible on Mac OS X and not other platforms, likely due to a different version of git

--- a/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommand.java
+++ b/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommand.java
@@ -251,7 +251,7 @@ public class GitChangeLogCommand extends AbstractChangeLogCommand implements Git
         SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);
         dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
 
-        Commandline cl = GitCommandLineUtils.getBaseGitCommandLine(workingDirectory, "whatchanged");
+        Commandline cl = GitCommandLineUtils.getBaseGitCommandLine(workingDirectory, "log");
         cl.createArg().setValue("--format=medium");
         cl.createArg().setValue("--decorate=short");
         cl.createArg().setValue("--raw");

--- a/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommandTest.java
+++ b/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommandTest.java
@@ -55,7 +55,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 (Date) null,
                 (Date) null,
                 40,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso --max-count=40 -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso --max-count=40 -- .");
     }
 
     @Test
@@ -65,7 +65,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 (Date) null,
                 (Date) null,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso -- .");
     }
 
     @Test
@@ -78,7 +78,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 startDate,
                 endDate,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 00:00:00 +0000\" \"--until=2007-10-10 00:00:00 +0000\" --date=iso -- .");
+                "git log --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 00:00:00 +0000\" \"--until=2007-10-10 00:00:00 +0000\" --date=iso -- .");
     }
 
     @Test
@@ -90,7 +90,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 startDate,
                 null,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" --date=iso -- .");
+                "git log --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" --date=iso -- .");
     }
 
     @Test
@@ -103,7 +103,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 startDate,
                 endDate,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" \"--until=2005-11-13 23:23:23 +0000\" --date=iso -- .");
+                "git log --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" \"--until=2005-11-13 23:23:23 +0000\" --date=iso -- .");
     }
 
     @Test
@@ -118,7 +118,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 endDate,
                 new ScmRevision("1"),
                 new ScmRevision("10"),
-                "git whatchanged --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" \"--until=2005-11-13 23:23:23 +0000\" --date=iso 1..10 -- .");
+                "git log --format=medium --decorate=short --raw --no-merges \"--since=2003-09-10 01:01:01 +0000\" \"--until=2005-11-13 23:23:23 +0000\" --date=iso 1..10 -- .");
     }
 
     @Test
@@ -131,7 +131,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 null,
                 endDate,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges \"--until=2003-11-10 00:00:00 +0000\" --date=iso -- .");
+                "git log --format=medium --decorate=short --raw --no-merges \"--until=2003-11-10 00:00:00 +0000\" --date=iso -- .");
     }
 
     @Test
@@ -141,7 +141,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 new ScmBranch("my-test-branch"),
                 (Date) null,
                 (Date) null,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso my-test-branch -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso my-test-branch -- .");
     }
 
     @Test
@@ -151,7 +151,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 new ScmRevision("1"),
                 null,
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso 1.. -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso 1.. -- .");
     }
 
     @Test
@@ -161,7 +161,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 new ScmRevision("1"),
                 new ScmRevision("10"),
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso 1..10 -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso 1..10 -- .");
     }
 
     @Test
@@ -171,7 +171,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 null,
                 new ScmRevision("1"),
                 new ScmRevision("1"),
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso 1..1 -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso 1..1 -- .");
     }
 
     @Test
@@ -181,7 +181,7 @@ public class GitChangeLogCommandTest extends ScmTestCase {
                 new ScmBranch("my-test-branch"),
                 new ScmRevision("1"),
                 new ScmRevision("10"),
-                "git whatchanged --format=medium --decorate=short --raw --no-merges --date=iso 1..10 my-test-branch -- .");
+                "git log --format=medium --decorate=short --raw --no-merges --date=iso 1..10 my-test-branch -- .");
     }
 
     // ----------------------------------------------------------------------

